### ISSUE TITLE
Code samples: Creating property editors programmatically

### ISSUE THREAD

**Issue by sofietoft** [umbraco/UmbracoDocs#2714]
In the Getting Started/Backoffice section of the docs, we have articles for most of the property editors in Umbraco. This documentation currently includes details on the implementation part.

We would like to have examples of **how to create each of the property editors programmatically** as well.
Some are simple and straightforward; others might be a bit more complicated.

[Link to examples will be posted as soon as possible.]

Property Editors that need code samples:
----
- [ ] Checkbox (Umbraco.TrueFalse)
- [ ] Checkbox list (Umbraco.CheckBoxList)
- [ ] Color Picker (Umbraco.ColorPicker)
- [ ] Content Picker (Umbraco.ContentPicker)
- [ ] Date (Umbraco.Date)
- [ ] Date/Time (Umbraco.DateTime)
- [ ] Decimal (Umbraco.Decimal)
- [ ] Dropdown (Umbraco.DropDown.Flexible)
- [ ] Email address (Umbraco.EmailAddress)
- [ ] File upload (Umbraco.UploadField)
- [ ] Grid layout (Umbraco.Grid)
- [ ] Image Cropper (Umbraco.ImageCropper)
- [ ] Label (Umbraco.NoEdit)
- [ ] List view (Umbraco.ListView)
- [ ] Markdown editor (Umbraco.MarkdownEditor)
- [ ] Media Picker (Umbraco.MediaPicker)
- [ ] Member Group Picker (Umbraco.MemberGroupPicker)
- [ ] Member Picker (Umbraco.MemberPicker)
- [ ] Multi Url Picker (Umbraco.MultiUrlPicker)
- [ ] Multinode Treepicker (Umbraco.MultiNodeTreePicker)
- [ ] Nested Content (Umbraco.NestedContent)
- [ ] Numeric (Umbraco.Integer)
- [ ] Radio button list (Umbraco.RadioButtonList)
- [ ] Repeatable textstrings (Umbraco.MultipleTextstring)
- [ ] Rich Text Editor (Umbraco.TinyMCEv3)
- [ ] Slider (Umbraco.Slider)
- [ ] Tags (Umbraco.Tags)
- [ ] Textarea (Umbraco.TextArea)
- [ ] Textbox (Umbraco.TextBox)
- [ ] User picker (Umbraco.UserPicker)

**Comment by sahil9001** [comment]
@sofietoft  I would like to work on one of the feature but I won't be able to complete without some assistance, expecting cooperation from your side! Thanks!

**Comment by Hard-Coder05** [comment]
@sofietoft 
I would like to work on this issue(one of the above property editors) as soon as you post the examples of what is to be done!

**Comment by sofietoft** [comment]
Thanks for checking in here!

@sahil9001 - I'd be happy to help out, if you have any questions. Just let me know!

@Hard-Coder05 - We're working on putting together an example. If you've worked with property editors programmatically before, you're welcome to come with a suggestion? :)

**Comment by sahil9001** [comment]
@sofietoft I would want an example to refer to, would be grateful if you could provide one!

**Comment by sofietoft** [comment]
Ah, definitely @sahil9001 

We are looking into providing an example here. Should be up before the end of the week.

**Comment by erikjanwestendorp** [comment]
@sofietoft I would like to work on this issue. I think an example as below will do what do you think? Do you have any suggestions for a scenario where people want to create their datatypes programmatically?

```csharp
var dataTypeService = Services.DataTypeService;

var trueFalseEditor = new TrueFalsePropertyEditor(Logger);
var dataTypeDefinition = new DataType(trueFalseEditor)
    {
        Name = "True/False - Default true",
        Configuration = new TrueFalseConfiguration
        {
            Default = true,
            ShowLabels = true,
            LabelOn = "On",
            LabelOff = "Off",
        }
    };

dataTypeService.Save(dataTypeDefinition);
```

**Comment by sofietoft** [comment]
Great to hear that you're up for the task @erikjanwestendorp ! 👏 

I'll run your questions by the Curator team, as I'll need some more technical backing to answer them 😉 
Regarding *where* to add these samples - we're thinking to add them as a new *section* at the bottom of the existing articles.

**Comment by sofietoft** [comment]
Oh, before I do ..
The sample you've pasted in here - that one is creating the property itself, right? 🤔 

What we were going for here, are samples on how to *add values to properties programmatically*.
That might not be entirely clear from the description on this issue - sorry for that!! I'll make sure it's corrected.

**Comment by erikjanwestendorp** [comment]
@sofietoft Yes my example creates the property itself indeed.. My bad.. Will an example as below make sense?

```csharp
var contentService = Services.ContentService;

var content = _contentService.GetById(1531);
content.SetValue("myCheckBox", true);
            
contentService.SaveAndPublish(content);
```

**Comment by marcemarc** [comment]
@erikjanwestendorp 

Yes, that's the kind of example we were thinking.

It's a common topic that crops up on the forums, particularly for the JSON based property editors!

So we thought it would be great to just add a snippet for how to set them to each property editor, and yes, we thought it could perhaps be the 'last thing' on each property editor page

regards

Marc
